### PR TITLE
Feat/timestamp block

### PR DESCRIPTION
**Pull request by frrist**
### What
This PR adds the timestamp field to block. This PR must be merged _after_ #2914

### CODE REVIEW

File: types/tipset_test.go [hunk 1]
```diff
@@ -153,7 +154,12 @@ func TestTipSet(t *testing.T) {
 		// String shouldn't really need testing, but some existing code uses the string as a
 		// datastore key and depends on the format exactly.
 		assert.Equal(t, "{ "+b1.Cid().String()+" }", RequireNewTipSet(t, b1).String())
-		assert.Equal(t, "{ "+b1.Cid().String()+" "+b2.Cid().String()+" "+b3.Cid().String()+" }",
+
+		// DONOTMETGE the below assertion needed to be fixed after adding a timestamp to block.
```
Comment (frrist) [inline types/tipset_test.go]: @anorth could you confirm my suspicion here, blame shows you as the author. Does the order of these matter?

File: plumbing/clock/testing.go [hunk 2]
```diff
@@ -0,0 +1,52 @@
+package clock
+
+import (
+	"time"
+
+	"github.com/benbjohnson/clock"
```
Comment (ZenGround0) [inline plumbing/clock/testing.go]: I noticed from looking that the issues that this isn't actively maintained and has some open issues about races (they might actually be addressed but issues are open).  Probably fine but something to keep in mind if we start seeing races in tests using it.

Comment (ZenGround0) [reply plumbing/clock/testing.go]: Looks like the de-racing PR got closed down without merging and the author recommended this project: https://github.com/jonboulle/clockwork.

Comment (frrist) [reply plumbing/clock/testing.go]: Ohh nice find! I remember looking at that issue but didn't see the recommendation, will address and bubble up the changes to the validation branch.

File: types/tipset_test.go (hunk 1 above)
Comment (anorth) [reply types/tipset_test.go]: Yes, order matters (from SortedCidSet)

Comment (anorth) [reply types/tipset_test.go]: You can robustify the test by actually using a sortedcidset to order them.

Comment (ingar) [reply types/tipset_test.go]: This is fixed in #2908

File: plumbing/clock/testing.go [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+// MockBlockClock defines a testing structure used to mock a BlockClock.
+type MockBlockClock struct {
+	blockClock *clock.Mock
+	blockTime  time.Duration
+}
+
+// NewMockBlockClock returns a new MockBlockClock with the provided blockTime.
+// Calls to MockBlockClock.EpochSeconds() will return 0 until a time is set using
+// the method SetClock().
+func NewMockBlockClock(blkTime time.Duration) *MockBlockClock {
```
Comment (anorth) [inline plumbing/clock/testing.go]: I suggest taking the mock time value in this constructor, saving most users a call to `SetClock`

**Pull request merged**